draft-plugin/server.go

```
@@ -0,0 +1,200 @@
+	log            *logrus.Entry
+	repos          []github.Repo
+	prowconfig     string
+	prowjob        string
+	regex          string
+	ns             string
+}
+
+type githubClient interface {
+	GetRef(org, repo, ref string) (string, error)
+}
+
+type prowJobClient interface {
+	Create(context.Context, *prowapi.ProwJob, metav1.CreateOptions) (*prowapi.ProwJob, error)
+	List(ctx context.Context, opts metav1.ListOptions) (*prowapi.ProwJobList, error)
+	Update(context.Context, *prowapi.ProwJob, metav1.UpdateOptions) (*prowapi.ProwJob, error)
+}
+
+func (s *server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	logrus.Info("inside http server")
+	eventType, eventGUID, payload, ok, _ := github.ValidateWebhook(w, r, s.tokenGenerator)
+	if !ok {
+		return
+	}
+	logrus.Info(w, "Event received. Have a nice day.")
+	if err := s.handleEvent(eventType, eventGUID, payload); err != nil {
+		logrus.WithError(err).Error("Error parsing event.")
+	}
+}
+
+func (s *server) handleEvent(eventType, eventGUID string, payload []byte) error {
+	l := logrus.WithFields(
+		logrus.Fields{
+			"event-type":     eventType,
+			github.EventGUID: eventGUID,
+		},
+	)
+	_ = l
+	var c *config.Config
+	var p = getProwClient()
+	c, err := config.Load(s.prowconfig, s.prowjob, nil, "")
+	if err != nil {
+		logrus.Info(err)
+	}
+	var pr github.PullRequestEvent
+	if err := json.Unmarshal(payload, &pr); err != nil {
+		return err
+	}
+	org, repo, author, ref := orgRepoAuthor(pr.PullRequest)
+	baseSHA := ""
+	baseSHAGetter := func() (string, error) {
+		var err error
+		baseSHA, err = githubClient.GetRef(s.ghc, org, repo, "heads/"+ref)
+		if err != nil {
+			return "", fmt.Errorf("failed to get baseSHA: %w", err)
+		}
+		return baseSHA, nil
+	}
+	headSHAGetter := func() (string, error) {
+		return pr.PullRequest.Head.SHA, nil
+	}
+	presubmits, err := c.GetPresubmits(s.gc, org+"/"+repo, baseSHAGetter, headSHAGetter)
+	if err != nil {
+		fmt.Printf("server not responding %s", err.Error())
```

Shouldn't we return the error from here, rather than printing the error only.

---

return as an error now